### ISSUE TITLE
ENH: code in datasets.real can be heavily factorized

### ISSUE THREAD

**Issue by mathurinm** [QB3/sparse-ho#24]
You have
```
def get_leukemia(csr=False):
    X, y = load_libsvm("leu")
    X_train, X_test, y_train, y_test = train_test_split(
        X, y, test_size=0.33, random_state=42)
    X_train, X_val, y_train, y_val = train_test_split(
        X_train, y_train, test_size=0.5, random_state=42)

    if csr:
        X_train = X_train.tocsr()
        X_val = X_val.tocsr()
        X_test = X_test.tocsr()
    else:
        X_train = X_train.tocsc()
        X_val = X_val.tocsc()
        X_test = X_test.tocsc()

    return X_train, X_val, X_test, y_train, y_val, y_test


def get_real_sim(csr=False):
    X, y = load_libsvm("real-sim")
    X_train, X_test, y_train, y_test = train_test_split(
        X, y, test_size=0.33, random_state=42)
    X_train, X_val, y_train, y_val = train_test_split(
        X_train, y_train, test_size=0.5, random_state=42)
    if csr:
        X_train = X_train.tocsr()
        X_val = X_val.tocsr()
        X_test = X_test.tocsr()
    else:
        X_train = X_train.tocsc()
        X_val = X_val.tocsc()
        X_test = X_test.tocsc()
    return X_train, X_val, X_test, y_train, y_val, y_test


def get_news20(csr=False):
    X, y = load_libsvm("news20")
    X_train, X_test, y_train, y_test = train_test_split(
        X, y, test_size=0.33, random_state=42)
    X_train, X_val, y_train, y_val = train_test_split(
        X_train, y_train, test_size=0.5, random_state=42)
    if csr:
        X_train = X_train.tocsr()
        X_val = X_val.tocsr()
        X_test = X_test.tocsr()
    else:
        X_train = X_train.tocsc()
        X_val = X_val.tocsc()
        X_test = X_test.tocsc()
    return X_train, X_val, X_test, y_train, y_val, y_test
```

**Comment by mathurinm** [comment]
also @agramfort maybe it would be worth joining forces in the existing celer.datasets module, but I understand if you don't want to have a "fragile" dependency in here. Your 2 cents ?

**Comment by agramfort** [comment]
yes we can do better here as this code is also present in benchopt. We
should have a pypi project for libsvm data fetchers... for a rainy sunday
afternoon...

>

**Comment by mathurinm** [comment]
![image](https://user-images.githubusercontent.com/8993218/94785782-6b1bd880-03d0-11eb-89fb-fdbe90236dbc.png)

**Comment by mathurinm** [comment]
closed in #27
